packages/flutter/lib/src/widgets/text_editing_action.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+/// The signature of a callback accepted by [TextEditingAction].
+typedef _OnInvokeTextEditingCallback<T extends Intent> = Object? Function(
+  T intent,
+  EditableTextState editableTextState,
+);
+
+/// An [Action] related to editing text.
+///
+/// If an [EditableText] is currently focused, the given [onInvoke] callback
+/// will be called with the [EditableTextState]. If not, then [isEnabled] will
+/// be false and [onInvoke] will not be called.
+///
+/// The focused [EditableText] must have a [Key]. This is handled automatically
+/// by built-in text editing widgets like [TextField], [CupertinoTextField],
+/// and [SelectableText], but for custom usage of [EditableText], it is
+/// necessary to explicitly pass in a key.
```

Comment:
Any concern about this?

I needed a way to get the EditableTextState and RenderEditable from the actions.  The way I chose to do it was by looking at the primaryFocus.  See the `_editableTextState` getter below.

Reply:
I had to add a `key` param to some EditableText tests below because of this.

Reply:
left a comment about this below.